backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/FormComponent.java

```diff
@@ -45,10 +47,29 @@ public Element render() {
         String targetObjectId = formDescription.getTargetObjectIdProvider().apply(variableManager);
         List<PageDescription> pageDescriptions = formDescription.getPageDescriptions();
 
+        List<Object> candidates = new ArrayList<>();
+
+        var optionalSelf = variableManager.get(VariableManager.SELF, Object.class);
+        if (optionalSelf.isPresent()) {
+            Object self = optionalSelf.get();
+            if (self instanceof Collection<?>) {
+                Collection<?> objects = (Collection<?>) self;
+                candidates.addAll(objects);
+            } else {
+                candidates.add(self);
+            }
+        }
+
         // @formatter:off
-        List<Element> children = pageDescriptions.stream().map(pageDescription -> {
-            PageComponentProps pageComponentProps = new PageComponentProps(variableManager, pageDescription);
-            return new Element(PageComponent.class, pageComponentProps);
+        List<Element> children = candidates.stream().map(candidate -> {
+            VariableManager childVariableManager = variableManager.createChild();
+            childVariableManager.put(VariableManager.SELF, candidate);
+            return childVariableManager;
+        }).flatMap(childVariableManager -> {
+            return pageDescriptions.stream().map(pageDescription -> {
+                PageComponentProps pageComponentProps = new PageComponentProps(childVariableManager, pageDescription);
+                return new Element(PageComponent.class, pageComponentProps);
+            });
```

Comment:
There is something about the formatting, I cannot say exactly what. The formatting does not look clean.

Reply:
The formatting matches what we do most of the time (here, it's the flatMap with the sub stream inside which can make it look odd). I can remove an useless `return` and simplify the indentation.